experimental/traffic-portal/src/app/api/testing/cache-group.service.ts

```diff
@@ -584,4 +593,39 @@ export class CacheGroupService {
 		}
 		return this.regions.splice(index, 1)[0];
 	}
+
+	public async getASNs(): Promise<Array<ResponseASN>>;
+	public async getASNs(id: number): Promise<ResponseASN>;
+
+	/**
+	 * Gets an array of ASNs from Traffic Ops.
+	 *
+	 * @param id If given, returns only the asn with the given id (number).
+	 * @returns An Array of ASNs objects - or a single ASN object if 'id'
+	 * was given.
+	 */
+	public async getASNs(id?: number): Promise<Array<ResponseASN> | ResponseASN> {
+		if(id) {
+			const asn = this.asns.find(a=>a.id === id);
+			if (!asn) {
+				throw new Error(`no such asn with id: ${id}`);
+			}
+			return asn;
+		}
+		return this.asns;
+	}
+
+	/**
+	 * Deletes an existing asn.
+	 *
+	 * @param asn Id of the asn to delete.
+	 * @returns The deleted asn.
+	 */
+	public async deleteASN(asn: number): Promise<ResponseASN> {
```

Comment:
This call signature is incompatible with the concrete service; it needs to also be able to accept a `ResponseASN` because things that are using the real service should be able to use this in the same way.

I'm working on a changeset that will enforce this at compile-time, which should hopefully make that easier to line up in the future.

Reply:
Ok, I was looking at region as an example and I think it is missing there too. I added it.